.github/workflows/build-and-test.yml

```
@@ -0,0 +1,38 @@
+name: Build and test checks
+on:
+  pull_request:
+    branches: ["*"]
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup pnpm
+        uses: pnpm/action-setup@v4
+        with:
+          version: 9
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 22
+          cache: "pnpm"
+
+      - name: Install dependencies
+        run: pnpm install --frozen-lockfile
+
+      - name: Create env file
+        run: cp .env.sample .env.local
+
+      - name: Lint
+        run: pnpm lint
+
+      - name: Build
+        run: pnpm build
+
+      - name: Test
+        run: pnpm test
+
+      - name: Test Coverage
+        run: pnpm test:coverage
```

Shouldn't we run tests before the build step to fail fast? Is there a specific reason for running tests after building?

---

Thanks for highlighting! I'll update the workflow to move the test step before the build step to follow the fail-fast strategy.